tooling/strace_analysis.sh

```diff
@@ -0,0 +1,267 @@
+        else
+            otherFiles+=("$file")
+            echo "No usrLocalFile: $file"
+        fi
+    done
+
+    echo "Number of /usr/local files: ${#usrLocalFiles[@]}"
+    echo "Number of other files: ${#otherFiles[@]}"
+}
+
+printNumberOfAllProcessedFiles() {
+    # Calculate and print number of all processed strace output files
+    totalLength=$((${#usrLocalFiles[@]} + ${#otherFiles[@]}))
+    if [ $totalLength -ne 0 ]; then
+        printf '\nNumber of all processed strace output files: %s\n' "$totalLength"
+    else
+        printf "\nNo strace output files available\n"
+        exit 1
+    fi
+}
+
+processFiles() {
+    for file in "${otherFiles[@]}"; do
+        echo "Processing: $file"
+
+        filePath="$(readlink -f "$file")"
+        pkg=$(rpm -qf "$filePath")
+
+        rc=$?
+        if [[ "$rc" != "0" ]]; then
+            # bin, lib, sbin pkgs may be installed under the root symlink
+            if [[ "$isBinSymLink" == "true" ]] && [[ $filePath == /usr/bin* ]]; then
+                filePath=${filePath/#\/usr\/bin/}
+                filePath="/bin${filePath}"
+                pkg=$(rpm -qf "$filePath" 2>/dev/null)
+                rc=$?
+            fi
+            if [[ "$isLibSymLink" == "true" ]] && [[ $filePath == /usr/lib* ]]; then
+                filePath=${filePath/#\/usr\/lib/}
+                filePath="/lib${filePath}"
+                pkg=$(rpm -qf "$filePath" 2>/dev/null)
+                rc=$?
+            fi
+            if [[ "$isSbinSymLink" == "true" ]] && [[ $filePath == /usr/sbin* ]]; then
+                filePath=${filePath/#\/usr\/sbin/}
+                filePath="/sbin${filePath}"
+                pkg=$(rpm -qf "$filePath" 2>/dev/null)
+                rc=$?
+            fi
+        fi
+
+        ignoreFile=false
+        for ignoreFile in "${ignores[@]}"; do
+            if [[ "$filePath" =~ $ignoreFile ]]; then
+                ignoreFile=true
+                break
+            fi
+        done
+        if [[ $ignoreFile == true ]]; then
+            continue
+        fi
+
+        if [[ "$rc" != "0" ]]; then
+            #echo "no pkg: $filePath"
+            usrLocalFiles+=("$filePath")
+        else
+            pkg="$(echo "$pkg" | cut -d" " -f1)"
+            pkg=${pkg::-1}
+            pkgString="pkg: $pkg version: $pkg"
+
+            if ! echo "${pkgs[@]-}" | grep "temurin_${pkgString}_temurin" >/dev/null; then
+                addSBOMFormulationComponentProperty "${javaHome}" "${classpath}" "${sbomJson}" "Build Dependencies" "Build tool package dependencies" "${pkg}" "${pkg}"
+                pkgs+=("temurin_${pkgString}_temurin")
+            fi
+        fi
+    done
+}
+
+processUsrLocalFiles() {
+    # loop over all non-package dependencies and try to get the version. If version is not empty, add to array
+    for file in "${usrLocalFiles[@]-}"; do
+        npkg=$("$file" --version 2>/dev/null | head -n 1)
+
+        if [[ "$npkg" != "" ]]; then
```

Comment:
I suspect this is not working for all the usr local files, eg.I am not seeing a version for "/usr/bin/sed" ?

Reply:
For /usr/bin/sed it produces `sed-4.2.2-7.el7.x86_6`. Is this not correct?

Reply:
ah yeah, I must be blind !